jobs/jobs_api_go_sdk.go

```diff
@@ -0,0 +1,289 @@
+	}
+	return noopLifecycleManager{}
+}
+
+type alwaysRunningLifecycleManagerGoSdk struct {
+	d *schema.ResourceData
+	m *common.DatabricksClient
+}
+
+func (a alwaysRunningLifecycleManagerGoSdk) OnCreate(ctx context.Context) error {
+	w, err := a.m.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+	jobID, err := parseJobId(a.d.Id())
+	if err != nil {
+		return err
+	}
+
+	return Start(jobID, a.d.Timeout(schema.TimeoutCreate), w, ctx)
+}
+
+func (a alwaysRunningLifecycleManagerGoSdk) OnUpdate(ctx context.Context) error {
+	w, err := a.m.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+	jobID, err := parseJobId(a.d.Id())
+	if err != nil {
+		return err
+	}
+
+	err = StopActiveRun(jobID, a.d.Timeout(schema.TimeoutUpdate), w, ctx)
+
+	if err != nil {
+		return err
+	}
+	return Start(jobID, a.d.Timeout(schema.TimeoutUpdate), w, ctx)
+}
+
+type controlRunStateLifecycleManagerGoSdk struct {
+	d *schema.ResourceData
+	m *common.DatabricksClient
+}
+
+func (c controlRunStateLifecycleManagerGoSdk) OnCreate(ctx context.Context) error {
+	return nil
+}
+
+func (c controlRunStateLifecycleManagerGoSdk) OnUpdate(ctx context.Context) error {
+	if c.d.Get("continuous") == nil {
+		return nil
+	}
+
+	jobID, err := parseJobId(c.d.Id())
+	if err != nil {
+		return err
+	}
+
+	w, err := c.m.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+
+	// Only use RunNow to stop the active run if the job is unpaused.
+	pauseStatus := c.d.Get("continuous.0.pause_status").(string)
+	if pauseStatus == "UNPAUSED" {
+		// Previously, RunNow() was not supported for continuous jobs. Now, calling RunNow()
+		// on a continuous job works, cancelling the active run if there is one, and resetting
+		// the exponential backoff timer. So, we try to call RunNow() first, and if it fails,
+		// we call StopActiveRun() instead.
+		_, err := w.Jobs.RunNow(ctx, jobs.RunNow{
+			JobId: jobID,
+		})
+
+		if err == nil {
+			return nil
+		}
+
+		// RunNow() returns 404 when the feature is disabled.
+		var apiErr *apierr.APIError
+		if errors.As(err, &apiErr) && apiErr.StatusCode != 404 {
+			return err
+		}
+	}
+
+	return StopActiveRun(jobID, c.d.Timeout(schema.TimeoutUpdate), w, ctx)
+}
+
+// Removing AutoterminationMinutes out of ClusterSpec's ForceSendFields because the current terraform plugin sdk
+// has a limitation that it will always set unspecified values to the zero value.
+//
+// `autotermination_minutes` should never be explicitly set for jobs because automated clusters don't accept this config.
+// TODO: consider removing `autotermination_minutes` from the jobs terraform schema.
+func removeAutoterminationMinutesFromForceSendFields(clusterSpec *compute.ClusterSpec) error {
+	newForceSendFields := []string{}
+	for _, field := range clusterSpec.ForceSendFields {
+		if field != "AutoterminationMinutes" {
+			newForceSendFields = append(newForceSendFields, field)
+		}
+	}
+	clusterSpec.ForceSendFields = newForceSendFields
+	return nil
+}
```

Comment:
Another option would be to remove `autotermination_minutes` from the jobs schema, as it should never be set from the jobs side

Reply:
I think if all the clusters are automated for jobs resource, then we could remove autotermination_minutes from the schema.